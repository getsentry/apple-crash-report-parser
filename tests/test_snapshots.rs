// These tests require serde for insta
#![cfg(feature = "with_serde")]

use std::fs;

use apple_crash_report_parser::AppleCrashReport;

fn load_fixture(name: &str) -> String {
    fs::read_to_string(format!("tests/fixtures/{}.txt", name)).unwrap()
}

#[test]
fn test_bruno() {
    let fixture = load_fixture("bruno");
    let report: AppleCrashReport = fixture.parse().unwrap();
    insta::assert_yaml_snapshot!("bruno", &report);
}

#[test]
fn test_handcrafted() {
    let fixture = load_fixture("handcrafted");
    let report: AppleCrashReport = fixture.parse().unwrap();
    insta::assert_yaml_snapshot!("handcrafted", &report);
}

#[test]
fn test_xcdyoutubekit_54() {
    let fixture = load_fixture("XCDYouTubeKit-54");
    let report: AppleCrashReport = fixture.parse().unwrap();
    insta::assert_yaml_snapshot!("XCDYouTubeKit-54", &report);
}
