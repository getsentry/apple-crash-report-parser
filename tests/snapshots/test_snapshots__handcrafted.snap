---
source: tests/test_snapshots.rs
expression: "&report"

---
incident_identifier: 5c32df84-31a0-43e7-87d0-239f7f594940
timestamp: "2019-01-09T17:42:22Z"
code_type: X86-64
path: /Users/bruno/Documents/Unreal Projects/YetAnotherMac/MacNoEditor/YetAnotherMac.app/Contents/MacOS/YetAnotherMac
application_specific_information: "objc_msgSend() selector name: respondsToSelector:\n  more information here"
report_version: 104
metadata:
  CrashReporter Key: TODO
  Exception Codes: SEGV_MAPERR at 0x88
  Exception Type: SIGSEGV
  Hardware Model: "MacBookPro14,3"
  Identifier: com.YourCompany.YetAnotherMac
  OS Version: Mac OS X 10.14.0 (18A391)
  Parent Process: "launchd [1]"
  Process: "YetAnotherMac [49028]"
  Version: 4.21.1
threads:
  - id: 0
    crashed: false
    frames:
      - module: libsystem_kernel.dylib
        instruction_addr: "0x7fff61bc6c2a"
      - module: CoreFoundation
        instruction_addr: "0x7fff349f505e"
      - module: CoreFoundation
        instruction_addr: "0x7fff349f45ad"
      - module: CoreFoundation
        instruction_addr: "0x7fff349f3ce4"
      - module: HIToolbox
        instruction_addr: "0x7fff33c8d895"
      - module: HIToolbox
        instruction_addr: "0x7fff33c8d5cb"
      - module: HIToolbox
        instruction_addr: "0x7fff33c8d348"
      - module: AppKit
        instruction_addr: "0x7fff31f4a95b"
      - module: AppKit
        instruction_addr: "0x7fff31f496fa"
      - module: AppKit
        instruction_addr: "0x7fff31f4375d"
      - module: YetAnotherMac
        instruction_addr: "0x108b7092b"
      - module: YetAnotherMac
        symbol: a_function_here
        instruction_addr: "0x108b702a6"
      - module: libdyld.dylib
        symbol: start
        instruction_addr: "0x7fff61a8e085"
      - module: YetanotherMac
        symbol: main
        filename: main.m
        lineno: 16
        instruction_addr: "0xea004"
  - id: 1
    name: Test Thread Name
    crashed: true
    frames:
      - module: libsystem_kernel.dylib
        instruction_addr: "0x7fff61bc85be"
      - module: libsystem_pthread.dylib
        instruction_addr: "0x7fff61c7f415"
      - instruction_addr: "0x54485244"
    registers:
      cs: "0x2b"
      fs: "0x0"
      gs: "0x0"
      r10: "0x0"
      r11: "0xffffffff"
      r12: "0x8"
      r13: "0x11e800b00"
      r14: "0x1"
      r15: "0x0"
      r8: "0x3"
      r9: "0x10"
      rax: "0x20261bb4775b008f"
      rbp: "0x700015a616d0"
      rbx: "0x0"
      rcx: "0x1288266c0"
      rdi: "0x0"
      rdx: "0x1"
      rflags: "0x10206"
      rip: "0x1090a0132"
      rsi: "0x0"
      rsp: "0x700015a613f0"
binary_images:
  - addr: "0x10864e000"
    size: 108797951
    uuid: 2d903291-397d-3d14-bfca-52c7fb8c5e00
    arch: x86_64
    version: 400.9.4 - 1.0.0
    name: YetAnotherMac
    path: /Users/bruno/Documents/Unreal Projects/YetAnotherMac/MacNoEditor/YetAnotherMac.app/Contents/MacOS/YetAnotherMac
  - addr: "0x112bb2000"
    size: 2170879
    uuid: 6deccee4-a052-3ea4-bb67-957b06f53ad1
    arch: x86_64
    version: 0.0.0 - 0.0.0
    name: libPhysX3PROFILE.dylib
    path: /Users/bruno/Documents/Unreal Projects/YetAnotherMac/MacNoEditor/YetAnotherMac.app/Contents/UE4/Engine/Binaries/ThirdParty/PhysX3/Mac/libPhysX3PROFILE.dylib
  - addr: "0x112fc0000"
    size: 221183
    uuid: 5e012a64-6cc5-36f1-9b4d-a0564049169b
    arch: x86_64
    version: 0.0.0 - 0.0.0
    name: libPhysX3CookingPROFILE.dylib
    path: /Users/bruno/Documents/Unreal Projects/YetAnotherMac/MacNoEditor/YetAnotherMac.app/Contents/UE4/Engine/Binaries/ThirdParty/PhysX3/Mac/libPhysX3CookingPROFILE.dylib
  - addr: "0x113013000"
    size: 1474559
    uuid: 9c198544-7194-3de6-b67e-4cc27eed2eab
    arch: x86_64
    version: 0.0.0 - 0.0.0
    name: libPhysX3CommonPROFILE.dylib
    path: /Users/bruno/Documents/Unreal Projects/YetAnotherMac/MacNoEditor/YetAnotherMac.app/Contents/UE4/Engine/Binaries/ThirdParty/PhysX3/Mac/libPhysX3CommonPROFILE.dylib
  - addr: "0x1131fa000"
    size: 28671
    uuid: 890f0997-f904-3544-9af7-cf011f09a06e
    arch: x86_64
    version: 400.9.0 - 1.0.0
    name: libPxFoundationPROFILE.dylib
    path: /Users/bruno/Documents/Unreal Projects/YetAnotherMac/MacNoEditor/YetAnotherMac.app/Contents/UE4/Engine/Binaries/ThirdParty/PhysX3/Mac/libPxFoundationPROFILE.dylib

